Request/ParamFetcher.php

```diff
@@ -195,25 +195,22 @@ public function cleanParamWithRequirements(Param $config, $param, $strict)
             );
         }
 
-        $this->checkNotIncompatibleParams($config);
```

Comment:
the fact that your PR removes this line makes it seem to me like maybe some of the changes you did are just the result of resolving some conflicts during a rebase?